src/modules/commander/commander.cpp

```diff
@@ -536,9 +537,16 @@ bool handle_command(struct vehicle_status_s *status_local, const struct safety_s
 		break;
 
 	case VEHICLE_CMD_COMPONENT_ARM_DISARM: {
+
+			//Refuse to arm if preflight checks have failed
```

Comment:
This check needs to be moved to this location a few lines below:
https://github.com/PX4/Firmware/pull/1895/files#diff-1d51a813dcb26f46616f71f28aacc70dL543

See the comment about in-air restore. If this is an in-air restore event, the checks should not be executed.